imagebuildah/stage_executor.go

```diff
@@ -1729,42 +1729,48 @@ func (s *StageExecutor) generateCacheKey(ctx context.Context, currNode *parser.N
 
 // cacheImageReference is internal function which generates ImageReference from Named repo sources
 // and a tag.
-func cacheImageReference(repo reference.Named, cachekey string) (types.ImageReference, error) {
-	tagged, err := reference.WithTag(repo, cachekey)
-	if err != nil {
-		return nil, fmt.Errorf("failed generating tagged reference for %q: %w", repo, err)
-	}
-	dest, err := imagedocker.NewReference(tagged)
-	if err != nil {
-		return nil, fmt.Errorf("failed generating docker reference for %q: %w", tagged, err)
+func cacheImageReferences(repos []reference.Named, cachekey string) ([]types.ImageReference, error) {
+	var result []types.ImageReference
+	for _, repo := range repos {
+		tagged, err := reference.WithTag(repo, cachekey)
```

Comment:
Wouldn't some of these be expected to fai, since the cachekey would only be in one of the repos?

Reply:
Yes, expected to fail in some of the repos. `--cache-to` and `--cache-from` will accept multiple repos. 

In case of `--cache-from` it will return if the key matched in first repo if not it will try for the other repo's in the list.

In case of `--cache-to` it will push the cache layers to all the repos in the list.